state.c

```diff
@@ -484,53 +484,75 @@ static int pv_state_start_platform(struct pv_state *s, struct pv_platform *p)
 int pv_state_run(struct pv_state *s)
 {
 	int ret = 0;
-	struct pv_platform *p, *tmp_p;
-
-	dl_list_for_each_safe(p, tmp_p, &s->platforms, struct pv_platform, list)
+	bool stop = false;
+	struct pv_group *g, *tmp_g;
+	dl_list_for_each_safe(g, tmp_g, &s->groups, struct pv_group, list)
```

Comment:
why do we need to iterate over groups here? can't we just move the login to pv_state_start_platform? and just initialized the timer of the group belonging to such platform if it wasn't already initialized when we start a platform? does this work for both update and non update bootup?

Reply:
This is the important change here... maybe it's only a misunderstood but AFAIK the groups has a dependency relationship between them in descendant order, i.e., the groups named first in the device.json must be processed before. Now based on that (and that could be the root of all the problem) you cannot iterate over the platforms, because the system not add them in the right order.

Reply:
That is not needed here. The group order works without your change, we are not introducing that feature here, as it is already working in master. It is implemented in pv_state_check_goals, where we iterate over groups and break if the group of the platform that we are checking are the same. This means we only check the previous groups, as they are ordered in memory. It's not beautiful, but adding another layer here is not going to help either.

So here, you only need to do what I mention in the comment. Check if the timer has been initialized when we start the platform.